pkg/services/rpcsrv/server.go

```diff
@@ -2729,29 +2729,24 @@ func (s *Server) subscribe(reqParams params.Params, sub *subscriber) (any, *neor
 		param := *p
 		jd := json.NewDecoder(bytes.NewReader(param.RawMessage))
 		jd.DisallowUnknownFields()
+		flt := neorpc.NewFilter(event)
+		err = flt.IsValid()
+		if err != nil {
+			return nil, neorpc.WrapErrorWithData(neorpc.ErrInvalidParams, err.Error())
+		}
+		err = jd.Decode(flt)
```

Comment:
You don't need this change, just leave this code as is and replace `var filter any` with `var filter NotificationFilter`. After decoding call `filter.IsValid`.